File: program/WordCount/ProcFunc/Top10WordList.rb
def getWordList
    arr=Array.new()
    File.foreach("news.txt") {|line|
        line.chomp.downcase.gsub(/[.,!'()-]/,"").split(/ +/).each {|word|
            arr.push word
        }
    }
    arr
end

def getWordCountList(arr)
    hash=Hash.new(0)
    arr.each {|word|
        hash[word]+=1
    }
    hash
end

def getTop10WordList(hash)
    arr=Array.new()
    strs=Array.new()
    hash.each {|k,v|
        arr.push [v, k]
    }
    arr.sort!{|a,b|(b[0]<=>a[0]).nonzero? || a[1]<=>b[1]}
    (0..[10,arr.length-1].min-1).each {|i|
        strs.push "#{arr[i][0]} #{arr[i][1]}"
    }
    strs
end

(getTop10WordList getWordCountList getWordList).each{|str|
    puts str
}

File: program/WordCount/ProcFunc/Top10WordList.pl
use strict;

print $_,"\n" for getTop10WordList(getWordCountList(getWordList()));

sub getWordList {
    my @list;
    open(FILE,"< news.txt");
    while(<FILE>){
        chomp;
        tr/A-Z/a-z/;
        tr/.,!'()-//d;
        push @list,split / +/
    }
    close(FILE);
    @list;
}

sub getWordCountList {
    my @list;
    my %words;
    for (@_) {
        $words{$_}++;
    }
    while(my @wc=each %words) {
        push @list,[@wc];
    }
    @list;
}

sub getTop10WordList {
    my @list;
    my @strList;
    my @list=sort {$b->[1]<=>$a->[1] or $a->[0] cmp $b->[0]} @_;
    my $count=@list;
    for (my $i=0;$i<$count && $i<10;$i++) {
        push @strList,sprintf("%d %s",$list[$i]->[1],$list[$i]->[0]);
    }
    @strList;
}
